akka-javasdk/src/main/java/akka/javasdk/http/QueryParams.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.http;
+
+
+import akka.http.javadsl.model.Query;
+import akka.japi.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents the query parameters of a request.
+ */
+public record QueryParams(Query query) {
```

Comment:
Make it an interface, not for user extension, with internal impl, for easier evolvability, like everything else in the SDK.

Not sure if there is any value in giving users access to the underlying Akka HTTP query when we have delegates for all use cases one could come up with.